mir/src/passes/inlining.rs

```diff
@@ -29,6 +29,59 @@ impl<'a> Inlining<'a> {
     pub fn new(diagnostics: &'a DiagnosticsHandler) -> Self {
         Self { diagnostics }
     }
+
+    /// Runs the Inlining pass once (with both InliningFirstPass and InliningSecondPass)
+    ///
+    /// Returns true if any calls were inlined, false otherwise, to let the caller know if any
+    /// changes were made or if we reached a fixed point.
+    fn run_once(&mut self, ir: &mut Mir) -> Result<bool, CompileError> {
+        // The first pass only identifies the call graph dependencies and the needed calls to inline
+        let mut first_pass = InliningFirstPass::new(self.diagnostics);
+        Visitor::run(&mut first_pass, ir.constraint_graph_mut())?;
+
+        // We then create the inlining order (inlining first the functions and evaluators that do
+        // not call other functions or evaluators)
+        let func_eval_inlining_order =
+            create_inlining_order(self.diagnostics, first_pass.func_eval_dependency_graph.clone())?;
+
+        // The second pass actually inlines the calls
+        let mut second_pass = InliningSecondPass::new(
+            self.diagnostics,
+            func_eval_inlining_order.clone(),
+            first_pass.func_eval_nodes_where_called.clone(),
+        );
+        Visitor::run(&mut second_pass, ir.constraint_graph_mut())?;
+
+        Ok(second_pass.had_calls)
+    }
+}
+
+// If we have to run the inlining algorithm `INLINING_LIMIT`, we return an error
+const INLINING_LIMIT: usize = 10;
```

Comment:
Where does this come from?

Reply:
Ah I see! I think my question was about why we chose 10 as a constant, though intuitively I imagine this is a good estimate.

Would it make sense to make this a variable parameter? Maybe through an environment variable?